src/lib/es5.d.ts

```diff
@@ -1043,14 +1043,14 @@ interface JSON {
       * @param replacer A function that transforms the results.
       * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
       */
-    stringify(value: any, replacer?: (this: any, key: string, value: any) => any, space?: string | number): string;
+    stringify(value: Object | string | number | boolean | symbol | null, replacer?: (this: any, key: string, value: any) => any, space?: string | number): string;

```

Comment:
nit: `JSON.stringify(Symbol.for('foo')) === undefined`, so `symbol` should be dropped

Reply:
problem is return type form stringfy. I think it should be undefined | string instead. This can be addressed by adding specific overloads. Please correct me if I am wrong.

Reply:
`string | undefined` with `--strictNullChecks` would force you make an existential check at runtime, even when it's statically provable what the return type would be

Reply:
but not when you have overloads.
```ts
stringify(value: Object | string | number | boolean  | null, replacer?: (this: any, key: string, value: any) => any, space?: string | number): string;

stringify(value: undefined | Function | symbol, replacer?: (this: any, key: string, value: any) => any, space?: string | number): undefined;
```

Reply:
i misunderstood; based on the type signature you wrote i thought you were advocating for a single api. my apologies!

you are correct, an overloaded api would be strictly better. i checked your definitions against [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Description) and they look correct too.

thank you :)

Reply:
Shouldn't it be `object` instead of `Object`?

And it would useful to move this into a `JSONStringifyable` type so it can be reused.

Reply:
you're correct; `object` narrows out `symbol` where `Object` does not:

```
declare class Foo {
	foo(o: object): boolean;
	foo(o: symbol | Function): string;
}
new Foo().foo(Symbol.for('')); // string

declare class Bar {
	bar(o: Object): boolean;
	bar(o: symbol | Function): string;
}
new Bar().bar(Symbol.for('')); // boolean
```

unfortunately, i found a different issue:

```
new Foo().foo(() => { }); // boolean, should be string
new Bar().bar(() => { }); // boolean, should be string
``` 

this would be most readily resolved by moving the overload with `Function` above the "happy path" overload. unsure if that's the right decision or if there's an alternative i'm missing.